include/blmc_robots/real_finger.hpp

```diff
@@ -373,4 +402,253 @@ robot_interfaces::finger::BackendPtr create_real_finger_backend(
+                    << " [A] Desired J1 "
+                    << ","
+                    << " [A] Desired J2 "
+                    << ","
+                    << " [A] Desired J3 "
+                    << ","
+                    << " Status " << std::endl;
+        output_file.close();
+
+        while (!destructor_was_called_ &&
+               !logger_data_->desired_action->wait_for_timeindex(0, 0.1))
+        // check wait_for_timeindex here !!
+        {
+        }
+
+        index = logger_data_->observation->newest_timeindex();
+
+        // std::cout << index << std::endl; //index is anywhere from 6 to 25
+        // (+/-), gen around 20
+
+        count = 0;
+        // meh used same index
+        for (long int t = 0; !destructor_was_called_; t++)
+        {
+            count++;
+
+            if (count == block_size_)
+            {
+                output_file.open("log.csv", std::ios_base::app);
+
+                auto t1 = std::chrono::high_resolution_clock::now();
+
+                for (int j = index; j < index + block_size_; j++)
+                {
+                    // std::cout.precision(dbl::max_digits10);
+                    // get returns the stored pointer
+                    output_file
+                        << std::fixed
+                        << logger_data_->observation->timestamp_s(j) << "," << j
+                        << "," << (*logger_data_->observation)[j].angle[0]
+                        << "," << (*logger_data_->observation)[j].angle[1]
+                        << "," << (*logger_data_->observation)[j].angle[2]
+                        << "," << (*logger_data_->observation)[j].velocity[0]
+                        << "," << (*logger_data_->observation)[j].velocity[1]
+                        << "," << (*logger_data_->observation)[j].velocity[2]
+                        << "," << (*logger_data_->observation)[j].torque[0]
+                        << "," << (*logger_data_->observation)[j].torque[1]
+                        << "," << (*logger_data_->observation)[j].torque[2]
+                        << "," << (*logger_data_->applied_action)[j][0] << ","
+                        << (*logger_data_->applied_action)[j][1] << ","
+                        << (*logger_data_->applied_action)[j][2] << ","
+                        << (*logger_data_->desired_action)[j][0] << ","
+                        << (*logger_data_->desired_action)[j][1] << ","
+                        << (*logger_data_->desired_action)[j][2] << ","
+                        << (*logger_data_->status)[j].action_repetitions
+                        << std::endl;
+
+                    // error here says trying to access more dimensions than are
+                    // there. Review eigen vectors. yeah, indexing begins from 0
+                    // :P
+
+                    std::cout << index << std::endl;
+                    std::cout << logger_data_->observation->newest_timeindex()
+                              << std::endl;
```

Comment:
These prints are only needed for debugging, right?  If yes, please either remove them or wrap them with a `#ifdef VERBOSE` if you think it could be useful in the future (in the latter case they would by default be disabled but could easily be enabled by adding `#define VERBOSE` at the top of the file).